--- a/include/net/netfilter/nf_conntrack.h
+++ b/include/net/netfilter/nf_conntrack.h
@@ -139,6 +139,16 @@ struct nf_conn {
 	/* Extensions */
 	struct nf_ct_ext *ext;

+	/*FanchmWrt Data*/
+	struct nf_fwx_data {
+		u32 mark;
+		u32 app_id;
+		u32 match_status;
+		u8 action;
+		u8 *host;
+		void *user_ptr;
+		struct list_head session_node;
+	} fwx_data;
 	/* Storage reserved for other modules, must be the last member */
 	union nf_conntrack_proto proto;
 };
--- /dev/null
+++ b/include/net/netfilter/nf_fwx_user.h
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+#ifndef _NF_FWX_USER_H
+#define _NF_FWX_USER_H
+
+struct sk_buff;
+struct nf_conn;
+
+struct nf_fwx_user_ops {
+	void (*conn_add)(struct nf_conn *ct);
+	void (*conn_del)(struct nf_conn *ct);
+};
+
+int nf_fwx_user_register_ops(const struct nf_fwx_user_ops *ops);
+void nf_fwx_user_unregister_ops(const struct nf_fwx_user_ops *ops);
+
+void nf_fwx_user_conn_add(struct nf_conn *ct);
+void nf_fwx_user_conn_del(struct nf_conn *ct);
+
+#endif
