--- a/net/netfilter/nf_conntrack_core.c	2026-06-08 17:08:30.914289030 +0800
+++ b/net/netfilter/nf_conntrack_core.c	2026-06-08 17:20:15.614122657 +0800
@@ -47,6 +47,7 @@
 #include <net/netfilter/nf_conntrack_timeout.h>
 #include <net/netfilter/nf_conntrack_labels.h>
 #include <net/netfilter/nf_conntrack_synproxy.h>
+#include <net/netfilter/nf_fwx_user.h>
 #include <net/netfilter/nf_nat.h>
 #include <net/netfilter/nf_nat_helper.h>
 #include <net/netns/hash.h>
@@ -101,6 +102,58 @@
 
 static struct conntrack_gc_work conntrack_gc_work;
 
+static DEFINE_MUTEX(nf_fwx_user_ops_lock);
+static const struct nf_fwx_user_ops __rcu *g_nf_fwx_user_ops = NULL;
+
+int nf_fwx_user_register_ops(const struct nf_fwx_user_ops *ops)
+{
+	if (!ops)
+		return -EINVAL;
+	printk("nf_fwx_user_register_ops: ops=%p\n", ops);
+	mutex_lock(&nf_fwx_user_ops_lock);
+	if (rcu_access_pointer(g_nf_fwx_user_ops)) {
+		mutex_unlock(&nf_fwx_user_ops_lock);
+		return -EBUSY;
+	}
+	rcu_assign_pointer(g_nf_fwx_user_ops, ops);
+	mutex_unlock(&nf_fwx_user_ops_lock);
+
+	return 0;
+}
+EXPORT_SYMBOL(nf_fwx_user_register_ops);
+
+void nf_fwx_user_unregister_ops(const struct nf_fwx_user_ops *ops)
+{
+	mutex_lock(&nf_fwx_user_ops_lock);
+	if (rcu_access_pointer(g_nf_fwx_user_ops) == ops)
+		RCU_INIT_POINTER(g_nf_fwx_user_ops, NULL);
+	mutex_unlock(&nf_fwx_user_ops_lock);
+	synchronize_rcu();
+}
+EXPORT_SYMBOL(nf_fwx_user_unregister_ops);
+
+void nf_fwx_user_conn_add(struct nf_conn *ct)
+{
+	const struct nf_fwx_user_ops *ops;
+
+	rcu_read_lock();
+	ops = rcu_dereference(g_nf_fwx_user_ops);
+	if (ops && ops->conn_add)
+		ops->conn_add(ct);
+	rcu_read_unlock();
+}
+
+void nf_fwx_user_conn_del(struct nf_conn *ct)
+{
+	const struct nf_fwx_user_ops *ops;
+
+	rcu_read_lock();
+	ops = rcu_dereference(g_nf_fwx_user_ops);
+	if (ops && ops->conn_del)
+		ops->conn_del(ct);
+	rcu_read_unlock();
+}
+
 void nf_conntrack_lock(spinlock_t *lock) __acquires(lock)
 {
 	/* 1) Acquire the lock */
@@ -1660,11 +1713,18 @@
 	memset_after(ct, 0, __nfct_init_offset);
 
 	nf_ct_zone_add(ct, zone);
+	/* fwx data init */
+	INIT_LIST_HEAD(&ct->fwx_data.session_node);
+	ct->fwx_data.user_ptr = NULL;
+
 
 	/* Because we use RCU lookups, we set ct_general.use to zero before
 	 * this is inserted in any list.
 	 */
 	refcount_set(&ct->ct_general.use, 0);
+	/* fwx user connect add */
+	nf_fwx_user_conn_add(ct);
+
 	return ct;
 out:
 	atomic_dec(&cnet->count);
@@ -1701,6 +1761,8 @@
 		rcu_read_unlock();
 	}
 
+	nf_fwx_user_conn_del(ct);
+
 	kfree(ct->ext);
 	kmem_cache_free(nf_conntrack_cachep, ct);
 	cnet = nf_ct_pernet(net);
