--- a/net/netfilter/nf_conntrack_standalone.c	2026-06-08 17:08:30.914289030 +0800
+++ b/net/netfilter/nf_conntrack_standalone.c	2026-06-08 17:20:22.550121020 +0800
@@ -363,6 +363,25 @@
 	seq_printf(s, "mark=%u ", READ_ONCE(ct->mark));
 #endif
 
+	{
+		struct fwx_user_node_view {
+			struct hlist_node mac_hnode;
+			rwlock_t lock;
+			u8 mac[ETH_ALEN];
+		};
+		struct fwx_user_node_view *user;
+
+		user = (struct fwx_user_node_view *)ct->fwx_data.user_ptr;
+		if (user)
+			seq_printf(s, "user_mac=%pM ", user->mac);
+	}
+
+
+	seq_printf(s, "appid=%u action=%u match_status=%u ",
+		   READ_ONCE(ct->fwx_data.app_id),
+		   READ_ONCE(ct->fwx_data.action),
+		   READ_ONCE(ct->fwx_data.match_status));
+
 	ct_show_secctx(s, ct);
 	ct_show_zone(s, ct, NF_CT_DEFAULT_ZONE_DIR);
 	ct_show_delta_time(s, ct);
